### PR TITLE
Extract ORMs from doorkeeper, take 2

### PR DESCRIPTION
**Pull request by tute**
Related PR: https://github.com/doorkeeper-gem/doorkeeper-orms/pull/2

Idea:

* Clone down and checkout doorkeeper when `doorkeeper-orms` runs `rake`
* Configure doorkeeper to load orm's code and each of its ORMs, run
  doorkeeper tests after extending with ORMs

To do:

- [x] Move documentation removed from here into doorkeeper-orms project
- [x] `doorkeeper-orms` renamed to `doorkeeper-mongodb`
- [x] Move ORM specific generators (might be buggy)
- [x] Move ORM specific code from`spec/*` to `doorkeeper-mongodb`.
- [ ] If successful, release doorkeeper 3, document in both projects. Requires an app making sure documentation is correct.

Both projects see a good portion of their code bases removed, keeping same behavior, test coverage, and getting faster specs. WIN.

### CODE REVIEW

**Comment by tute** [comment]
@jasl please tell me how you like these changes! I think this is a good way of solving this problem.

**Comment by jasl** [comment]
Let's me check after work.

**Comment by jasl** [comment]
@tute I comment in [doorkeeper-orms](https://github.com/doorkeeper-gem/doorkeeper-orms/pull/2/files#r29564161)

File: doorkeeper.gemspec [hunk 1]
```diff
@@ -26,6 +26,5 @@ Gem::Specification.new do |s|
   s.add_development_dependency "timecop", "~> 0.7.0"
   s.add_development_dependency "database_cleaner", "~> 1.3.0"
   s.add_development_dependency "rspec-activemodel-mocks", "~> 1.0.0"
-  s.add_development_dependency "bcrypt-ruby", "~> 3.0.1"
   s.add_development_dependency "pry", "~> 0.10.0"
```
Comment (tute) [inline doorkeeper.gemspec]: @jasl can we remove dependencies from here?

Comment (jasl) [inline doorkeeper.gemspec]: I think yes

**Comment by tute** [comment]
It looks both PRs are green. Updated documentation in both repos, and renamed the mongodb extensions one.

We just need a real mongo doorkeeper application to make sure everything has been tied together correctly. Do you have anyone at hand, @jasl? Thank you!

**Comment by jasl** [comment]
No problem, I will update our application to doorkeeper edge version and test

**Comment by jasl** [comment]
We have update our developing branch using this gem, if have problem, I will fix

**Comment by jasl** [comment]
oh I mean doorkeeper-mongo

**Comment by tute** [comment]
I created a new `2.2-stable` branch that will receive security fixes from now on. master is already tracking version 3, which I expect to release whenever we see a mongodb doorkeeper v3 installation working correctly, and making sure documentation is correct.

Thank you!

**Comment by tute** [comment]
In preparation for Rails 3, if we still have changes we didn't do due to backwards compatibility concerns, or we have leftover upgrading warnings, we should change/remove those now.

**Comment by tute** [comment]
@jasl could you test doorkeeper 3 with your MongoDB project, so we make sure it works and can release v3 stable?

**Comment by jasl** [comment]
OK, give me a week to test it stable.

**Pull request merged**